packages/boot/test/acceptance/controller.booter.app.acceptance.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright IBM Corp. 2013,2018. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Client, createClientForHandler, TestSandbox} from '@loopback/testlab';
+import {RestServer} from '@loopback/rest';
+import {BootOptions} from '@loopback/core';
+import {ControllerBooter, BootComponent} from '../../index';
+import {resolve} from 'path';
+
+describe('controller booter acceptance tests', () => {
+  // tslint:disable-next-line:no-any
+  let app: any;
+  const SANDBOX_PATH = resolve(__dirname, '../sandbox');
+  const sandbox = new TestSandbox(SANDBOX_PATH);
+  const bootConfig = {
+    projectRoot: SANDBOX_PATH,
+  };
+
+  beforeEach(resetSandbox);
+  beforeEach(getApp);
+
+  afterEach(stopApp);
+
+  it('binds controllers using ControllerDefaults and REST endpoints work', async () => {
+    await app.booter(ControllerBooter);
+    await app.boot(bootConfig);
```

Comment:
Ditto, it is the `ControllerBootApp` that knows where it has files, not the code using the app!

I think you should either
 - move all bootstrapping logic to your custom Application class used by his test
 - or don't load the custom app from any file, either define the class here in the test, or don't use any class at all and simply call BootStrapper on a vanilla Application instance.